tools/delete_imaging_upload.pl

```diff
@@ -0,0 +1,1628 @@
+
+=head3 deleteUploadsOnFileSystem($filesRef, $scanTypesToDeleteRef, $keepDefaced)
+
+This method deletes from the file system all the files associated to the upload(s) passed on the
+command line that were found on the file system. A warning will be issued for any file that
+could not be deleted.
+
+INPUTS:
+  - $filesRef: reference to the array that contains the file informations for all the files
+    that are associated to the upload(s) passed on the command line.
+  - $scanTypesToDeleteRef: reference to the array that contains the list of scan type names to delete.
+  - $keepDefaced: whether the defaced files should be kept or not.
+               
+=cut
+sub deleteUploadsOnFileSystem {
+    my($filesRef, $scanTypesToDeleteRef, $keepDefaced) = @_;
+    
+    my %deletedFile;
+    foreach my $t (@PROCESSED_TABLES) {
+        foreach my $f (@{ $filesRef->{$t} }) {
+            next if !shouldDeleteFile($t, $f, $scanTypesToDeleteRef, $keepDefaced);
+            
+            next if !$f->{'Exists'};
+            
+            next if $deletedFile{ $f->{'FullPath'} };
+            
+            NeuroDB::MRI::deleteFiles($f->{'FullPath'});
+            $deletedFile{ $f->{'FullPath'} } = 1;
+        }
+    }
+}
+
+=pod
+
+=head3 getTypesToDelete($dbh, $scanTypeList, $keepDefaced)
+
+Gets the list of names of the scan types to delete. If C<-type> was used, then this list is built
+using the argument to this option. If C<-defaced> was used, then the list is fetched using the config
+setting C<modalities_to_deface>.
+
+INPUTS:
+  - $dbh: database handle.
+  - $scanTypeList: comma separated string of scan type names.
+  - $keepDefaced: whether the defaced files should be kept or not.
+                  
+RETURNS:
+  - A reference on a hash of the names of the scan types to delete: key => scan type name
+    value => 1 or 0 depending on whether the name is valid or not.
+=cut
+sub getScanTypesToDelete {
+    my($dbh, $scanTypeList, $keepDefaced) = @_;
+    
+    return () if !defined $scanTypeList && !$keepDefaced;
+    
+    if($keepDefaced) {
+        my $scanTypesRef = &NeuroDB::DBI::getConfigSetting(\$dbh, 'modalities_to_deface');
+        return defined $scanTypesRef ? (map { $_ => 1 } @$scanTypesRef) : ();
+    }
+    
+    my %types = map { $_=> 1 } split(/,/, $scanTypeList);
+    
+    $query = 'SELECT Scan_type, ID FROM mri_scan_type '
+           . 'WHERE Scan_type IN ('
+           . join(',', ('?') x keys %types)
+           . ')';
+    my $validTypesRef = $dbh->selectall_hashref($query, 'Scan_type', undef, keys %types);
+    
+    return map { $_ => defined $validTypesRef->{$_} } keys %types;
+}
+
+=pod
+
+=head3 getTarchiveSeriesIDs($dbh, $filesRef)
+
+Gets the list of C<TarchiveSeriesID> to delete in table C<tarchive_files>.
+
+INPUTS:
+  - $dbh: database handle.
+  - $filesRef: reference to the array that contains the file informations for all the files
+    that are associated to the upload(s) passed on the command line.
+           
```

Comment:
missing the tail of the documentation with authorship
```
=pod

=head1 LICENSING

License: GPLv3

=head1 AUTHORS

LORIS community <loris.info@mcin.ca> and McGill Centre for Integrative
Neuroscience

=cut
```

Reply:
OK, added.